Yosemite/YosemiteTests/Stores/NotificationCountStoreTests.swift

```diff
@@ -0,0 +1,166 @@
+
+    /// Mockup Storage: InMemory
+    ///
+    private var storageManager: MockupStorageManager!
+
+    /// Mockup File Storage: Load data in memory
+    ///
+    private var fileStorage: MockInMemoryStorage!
+
+    /// Test subject
+    ///
+    private var subject: NotificationCountStore!
+
+    private let defaultSiteID: Int64 = 134
+
+    override func setUp() {
+        super.setUp()
+        dispatcher = Dispatcher()
+        storageManager = MockupStorageManager()
+        fileStorage = MockInMemoryStorage()
+        subject = NotificationCountStore(dispatcher: dispatcher, storageManager: storageManager, fileStorage: fileStorage)
+    }
+
+    override func tearDown() {
+        dispatcher = nil
+        storageManager = nil
+        fileStorage = nil
+        subject = nil
+        super.tearDown()
+    }
+
+    // MARK: `incrementNotificationCount`
+
+    func testIncrementingAndLoadingNotificationCountReturnsTheCorrectCount() {
+        let data = NotificationCountPListWrapper(notificationCountBySite: [defaultSiteID: [.comment: 2, .storeOrder: 6]])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.incrementNotificationCount(siteID: defaultSiteID, type: .comment, incrementCount: 10) {
+                let loadAction = NotificationCountAction.loadNotificationCount(siteID: self.defaultSiteID, type: .comment) { count in
+                    notificationCount = count
+                    expectation.fulfill()
+                }
+                self.subject.dispatcher.dispatch(loadAction)
+            }
+            subject.dispatcher.dispatch(action)
+        }
+
+        XCTAssertEqual(notificationCount, 12)
+    }
+
+    // MARK: `loadNotificationCount`
+
+    func testLoadingNotificationCountWithoutPreviousDataReturns0() {
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.loadNotificationCount(siteID: defaultSiteID, type: .comment) { count in
+                notificationCount = count
+                expectation.fulfill()
+            }
+            subject.dispatcher.dispatch(action)
+        }
+
+        XCTAssertEqual(notificationCount, 0)
+    }
+
+    // MARK: `resetNotificationCount`
+
+    func testResettingNotificationCountOfAGivenTypeReturns0() {
+        let data = NotificationCountPListWrapper(notificationCountBySite: [defaultSiteID: [.comment: 2, .storeOrder: 6]])
+        try! fileStorage.write(data, to: URL(fileURLWithPath: ""))
+
+        var notificationCount: Int?
+        waitForExpectation { expectation in
+            let action = NotificationCountAction.resetNotificationCount(siteID: defaultSiteID, type: .comment) {
+                let loadAction = NotificationCountAction.loadNotificationCount(siteID: self.defaultSiteID, type: .comment) { count in
```

Comment:
This makes me wonder whether we should return the current count in the `resetNotificationCount` callback. 😄

Reply:
yea I thought about it, but we don't currently use the reset count in the app 😐 okay to keep it this way for now?